plugins/modules/opkg.py

```diff
@@ -89,112 +89,82 @@
-
-        remove_c += 1
-
-    if remove_c > 0:
-
-        module.exit_json(changed=True, msg="removed %s package(s)" % remove_c)
-
-    module.exit_json(changed=False, msg="package(s) already absent")
-
-
-def install_packages(module, opkg_path, packages):
-    """ Installs one or more packages if not already installed. """
-
-    p = module.params
-    force = p["force"]
-    if force:
-        force = "--force-%s" % force
-
-    install_c = 0
-
-    for package in packages:
-        if query_package(module, opkg_path, package) and (force != '--force-reinstall'):
-            continue
-
-        rc, out, err = module.run_command("%s install %s %s" % (opkg_path, force, package))
-
-        if not query_package(module, opkg_path, package):
-            module.fail_json(msg="failed to install %s: %s" % (package, out))
-
-        install_c += 1
-
-    if install_c > 0:
-        module.exit_json(changed=True, msg="installed %s package(s)" % (install_c))
-
-    module.exit_json(changed=False, msg="package(s) already present")
-
-
-def main():
-    module = AnsibleModule(
+class Opkg(StateModuleHelper):
+    module = dict(
         argument_spec=dict(
             name=dict(aliases=["pkg"], required=True, type="list", elements="str"),
             state=dict(default="present", choices=["present", "installed", "absent", "removed"]),
             force=dict(default="", choices=["", "depends", "maintainer", "reinstall", "overwrite", "downgrade", "space", "postinstall", "remove",
                                             "checksum", "removal-of-dependent-packages"]),
             update_cache=dict(default=False, type='bool'),
-        )
+        ),
     )
 
-    opkg_path = module.get_bin_path('opkg', True, ['/bin'])
+    def __init_module__(self):
+        self.vars.set("install_c", 0, output=False, change=True)
+        self.vars.set("remove_c", 0, output=False, change=True)
 
-    p = module.params
+        state_map = dict(
+            query="list-installed",
+            present="install",
+            installed="install",
+            absent="remove",
+            removed="remove",
+        )
+
+        def _force(value):
+            if value == "":
+                value = None
+            return cmd_runner_fmt.as_optval("--force-")(value)
+
+        self.runner = CmdRunner(
+            self.module,
+            command="opkg",
+            arg_fornats=dict(
+                package=cmd_runner_fmt.as_list(),
+                state=cmd_runner_fmt.as_map(state_map),
+                force=cmd_runner_fmt.as_func(_force),
+                update_cache=cmd_runner_fmt.as_bool("update")
+            ),
+            check_rc=True,
+        )
 
-    if p["update_cache"]:
-        update_package_db(module, opkg_path)
+    def _package_in_desired_state(self, name, desired_installed):
+        dummy, out, dummy = self.runner("state").run(state="query")
 
-    pkgs = p["name"]
+        has_package = any(s.startswith(name) for s in out.split("\n"))
```

Comment:
```suggestion
        has_package = out.startswith(name + " ")
```

Reply:
In the previous version there was a `grep` for a line starting with it. Does this mean that is the only line?

Reply:
If you add the name of the package as a second argument (`self.runner("state package")` -> command: `opkg list-installed PACKAGE`), yes.